keras_preprocessing/image.py

```diff
@@ -2102,26 +2102,17 @@ def __init__(self, dataframe, directory, image_data_generator,
             if not ext_exist:
                 raise ValueError('has_ext is set to True but'
                                  ' extension not found in x_col')
-            temp_df = pd.DataFrame({x_col: filenames}, dtype=str)
-            temp_df = self.df.merge(temp_df, how='right', on=x_col)
-            temp_df = temp_df.set_index(x_col)
-            temp_df = temp_df.reindex(filenames)
-            temp_df = temp_df.dropna()
-            self.filenames = list(temp_df.index)
+            self.df = self.df[self.df[x_col].isin(filenames)]
```

Comment:
`self.df = self.df[self.df[x_col].isin(filenames)].sort_values(by=x_col)` could you add this small change to your PR? This sorts the final dataframe and filenames.